internal/service/rds/cluster_activity_stream.go

```diff
@@ -0,0 +1,148 @@
+		CreateContext: resourceAwsRDSClusterActivityStreamCreate,
+		ReadContext:   resourceAwsRDSClusterActivityStreamRead,
+		DeleteContext: resourceAwsRDSClusterActivityStreamDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"kms_key_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"mode": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(rds.ActivityStreamMode_Values(), false),
+			},
+			"kinesis_stream_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"engine_native_audit_fields_included": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsRDSClusterActivityStreamCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).RDSConn
+
+	resourceArn := d.Get("resource_arn").(string)
+
+	startActivityStreamInput := &rds.StartActivityStreamInput{
+		ResourceArn:                     aws.String(resourceArn),
+		ApplyImmediately:                aws.Bool(true),
+		KmsKeyId:                        aws.String(d.Get("kms_key_id").(string)),
+		Mode:                            aws.String(d.Get("mode").(string)),
+		EngineNativeAuditFieldsIncluded: aws.Bool(d.Get("engine_native_audit_fields_included").(bool)),
+	}
+
+	log.Printf("[DEBUG] RDS Cluster start activity stream input: %s", startActivityStreamInput)
+
+	resp, err := conn.StartActivityStream(startActivityStreamInput)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating RDS Cluster Activity Stream: %s", err))
+	}
+
+	log.Printf("[DEBUG]: RDS Cluster start activity stream response: %s", resp)
+
+	d.SetId(resourceArn)
+
+	err = waitActivityStreamStarted(ctx, conn, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceAwsRDSClusterActivityStreamRead(ctx, d, meta)
+}
+
+func resourceAwsRDSClusterActivityStreamRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).RDSConn
+
+	log.Printf("[DEBUG] Finding DB Cluster (%s)", d.Id())
+	resp, err := FindDBClusterByClusterArn(conn, d.Id())
+
+	if err != nil {
+		if tfresource.NotFound(err) {
+			log.Printf("[WARN] RDS Cluster (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(fmt.Errorf("error describing RDS Cluster (%s): %s", d.Id(), err))
+	}
+
+	if resp == nil {
+		return diag.FromErr(fmt.Errorf("error retrieving RDS cluster: empty response for: %s", d.Id()))
+	}
+
+	if aws.StringValue(resp.ActivityStreamStatus) == rds.ActivityStreamStatusStopped {
```

Comment:
I think we should fold this into the find function (also will remove the dupe in acctest destory function)

Reply:
Do you suggest the `FindDBClusterByClusterArn` function or do you suggest I create a new find function for the Activity Stream? The Activity Stream isn't really something you can "find" and return. It's a status property on the DB cluster.